### PR TITLE
add payment FE

### PR DESCRIPTION
**Pull request by JunZheng007**
### What this PR does (required):
- make FE for the payment method on the setting page
- only limit one card for now

### Screenshots / Videos (front-end only):
image

### Any issues with the current functionality (optional):
- I can't customize the style of stripe Card Element.

### CODE REVIEW

File: client/src/components/PaymentMethods/PaymentCard/PaymentCard.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        return visaLogo;
+      case 'mastercard':
+        return mcLogo;
+    }
+  };
+
+  return (
+    <Card className={classes.paymentCard}>
+      <CardMedia style={{ width: '30%' }} component="img" image={getCardLogo(cardElement?.card?.brand)} />
+      <div>
```
Comment (ezeilo-su) [inline client/src/components/PaymentMethods/PaymentCard/PaymentCard.tsx:22]: Let's use MUI component where possible

**Review by ezeilo-su — COMMENTED** [review]
Nice job on this.

File: client/src/App.tsx [hunk 2]
```diff
@@ -21,6 +21,13 @@ import ProfileLists from './pages/ProfileList/profileList';
 import { LocalizationProvider } from '@mui/lab';
 import AdapterDateFns from '@mui/lab/AdapterDateFns';
 import { AuthRoute } from './components/AuthRoute/AuthRoute';
+import { loadStripe } from '@stripe/stripe-js';
+import { Elements } from '@stripe/react-stripe-js';
+
+const stripePromise = loadStripe(
```
Comment (rajivtitus) [inline client/src/App.tsx:27]: 👍🏽

File: client/src/components/PaymentMethods/useStyle.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Theme } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+
+export const useStyles = makeStyles((theme: Theme) => ({
+  card: {
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/components/PaymentMethods/useStyle.ts:6]: These styles can directly be added to the `Box` component as props

```
<Box display="flex alginItems="center"...> 
```

File: client/src/components/PaymentMethods/useStyle.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    alignSelf: 'center',
+    color: 'grey',
+  },
+
+  paymentCard: {
+    margin: '20px 5px',
+    width: '250px',
+    height: '150px',
+    padding: '15px',
+    display: 'flex',
```
Comment (rajivtitus) [inline client/src/components/PaymentMethods/useStyle.ts:31]: Same as comment above

File: client/src/components/PaymentMethods/PaymentMethods.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  const classes = useStyles();
+  const stripe = useStripe();
+  const elements = useElements();
+  const [cardElement, setCardElement] = useState<PaymentMethod>();
+  const [openAddPayment, setOpenAddPayment] = useState(false);
+
+  const handleSubmit = async (value: { name: string }) => {
+    if (stripe && elements) {
+      await stripe
+        .createPaymentMethod({
```
Comment (rajivtitus) [inline client/src/components/PaymentMethods/PaymentMethods.tsx:19]: Why don't we update the snackbar with an error message in the event of a promise rejection

**Review by rajivtitus — APPROVED** [review]
Good work Jun! 👍🏽 
Just a couple of small comments to address before merging, thanks!

File: client/src/components/PaymentMethods/useStyle.ts (hunk 3 above)
Comment (JunZheng007) [reply client/src/components/PaymentMethods/useStyle.ts:6]: Box only can take sx as props for CSS elements, so I think this way is easier for styling.

File: client/src/components/PaymentMethods/useStyle.ts (hunk 4 above)
Comment (JunZheng007) [reply client/src/components/PaymentMethods/useStyle.ts:31]: same reason as above

File: client/src/components/PaymentMethods/PaymentMethods.tsx (hunk 5 above)
Comment (JunZheng007) [reply client/src/components/PaymentMethods/PaymentMethods.tsx:19]: add it

**Pull request merged**